src/cpu/arm.rs

```diff
@@ -25,11 +25,62 @@ fn detect_features() -> u32 {
     use libc::c_ulong;
 
     // XXX: The `libc` crate doesn't provide `libc::getauxval` consistently
-    // across all Android/Linux targets, e.g. musl.
+    // across all Android/Linux targets, e.g. uclibc.
+    // If weak linking of extern symbols becomes available, this code could
+    // choose to dynamically fall back to the procfs version below.
+    #[cfg(not(target_env = "uclibc"))]
     extern "C" {
         fn getauxval(type_: c_ulong) -> c_ulong;
     }
 
+    #[cfg(not(target_env = "uclibc"))]
+    fn getauxval_safe(type_: c_ulong) -> c_ulong {
+        // safety: getauxval seems to have no safety concerns
+        unsafe { getauxval(type_) }
+    }
+
+    // Provide a compatible implementation for uclibc based on procfs.
+    #[cfg(all(target_env = "uclibc"))]
+    fn getauxval_safe(type_: c_ulong) -> c_ulong {
+        let fd = unsafe {
+            // safety: static string has interned NUL terminator
+            libc::open(b"/proc/self/auxv\0".as_ptr(), libc::O_RDONLY)
```

Comment:
We could just use libstd here since the target_os is linux and libstd is supported by this target. Thus we could avoid the `unsafe` code completely, right?

OTOH, we've traditionally gone to some lengths to avoid File I/O whenever practical in this library. Is there really no way to find auxv in memory for uClibc-ng?

Reply:
Yeah I originally did a libstd implementation, but then I guessed that it would be less preferred than one which didn't add any dependencies.  Would you prefer I submit my libstd implementation?

glibc uses some tight integration with the kernel's ELF loader to find a spot in memory where the aux vector has been stored away.  I could try to duplicate that here, but it felt much larger and more fragile, having looked at the recent addition of getauxval to uclibc [here](https://cgit.uclibc-ng.org/cgi/cgit/uclibc-ng.git/commit/?id=d869bb1600942c01a77539128f9ba5b5b55ad647).

I get the desire to avoid file IO.  Is it any consolation that no actual I/O device will be involved?  procfs just talks with the kernel, which just printfs into the user space buffer, so it's very fast.

I'm very happy to move this in whatever direction you prefer!